rockcraft/parts.py

```diff
@@ -67,6 +69,10 @@ def __init__(
 
         emit.progress("Initializing parts lifecycle")
 
+        if not PartsLifecycle._OVERLAY_CALLBACK_REGISTERED:
```

Comment:
this seems fragile, do want to have a dedicated class to keep track of registration?

Reply:
Updated this, please take a look